[Semantics] Return parent Location and Equipment

jimtng opened the pull request:
Resolve #2924 

Instead of returning the given item, getEquipment() and getLocation() return the item's parent Equipment / Location. This allows the caller to navigate the hierarchy, e.g. get the parent location of the given location.

jlaur commented on bundles/org.openhab.core.model.script/src.moved/test/java/org/openhab/core/model/script/actions/SemanticsTest.java:
```diff
@@ -95,8 +104,8 @@ public void setup() throws ItemNotFoundException {
     public void testGetLocation() {
-        assertThat(Semantics.getLocation(indoorLocationItem), is(indoorLocationItem));
-        assertThat(Semantics.getLocation(bathroomLocationItem), is(bathroomLocationItem));
+        assertNull(Semantics.getLocation(indoorLocationItem));
```
For consistency with the constraint-based Assert model:
```suggestion
        assertThat(Semantics.getLocation(indoorLocationItem), is(nullValue()));
```

jlaur commented on bundles/org.openhab.core.model.script/src.moved/test/java/org/openhab/core/model/script/actions/SemanticsTest.java:
```diff
@@ -115,7 +124,8 @@ public void testGetLocationType() {
     @Test
     public void testGetEquipment() {
-        assertThat(Semantics.getEquipment(equipmentItem), is(equipmentItem));
+        assertNull(Semantics.getEquipment(equipmentItem));
```
```suggestion
        assertThat(Semantics.getEquipment(equipmentItem), is(nullValue()));
```

jlaur reviewed (changes requested):
Nice! Two minor comments for the tests.

jimtng commented:
@jlaur thanks for your review. I have changed the assertion as suggested. I'm not sure about the spacing in-between lines and why they are that way, but I added spacing to make them all spaced the same way.

wborn reviewed (approved):
Thanks! It looks good to me, but it would be nice if @cweitkamp can also have a quick look.

cweitkamp reviewed (approved):
LGTM. Thanks.

It sounds reasonable to change the logic.
